database/migrations/1749227968774_alter_field_of_studies_and_add_study_types_table.ts

```diff
@@ -0,0 +1,71 @@
+import { BaseSchema } from "@adonisjs/lucid/schema";
+
+export default class extends BaseSchema {
+  protected tableName = "fields_of_studies";
+  protected studiesTypes = ["1DEGREE", "2DEGREE", "LONG_CYCLE"];
+
+  async up() {
+    await this.schema.raw(`
+      DO $$ BEGIN
+        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'studies_type') THEN
+          CREATE TYPE "studies_type" AS ENUM (${this.studiesTypes.map((type) => `'${type}'`).join(", ")});
+        END IF;
+      END $$;
+      DO $$ BEGIN
+        IF NOT EXISTS (
+            SELECT 1
+            FROM information_schema.columns
+            WHERE table_name = '${this.tableName}' AND column_name = 'studies_type'
+        ) THEN ALTER TABLE "${this.tableName}" ADD COLUMN "studies_type" "studies_type" NULL;
+        END IF;
+      END $$;
+    `);
+    await this.db.rawQuery(`
+      UPDATE ${this.tableName}
+      SET studies_type = CASE
+                   WHEN semester_count = 12 THEN 'LONG_CYCLE'::studies_type
+                   WHEN is_2nd_degree = true THEN '2DEGREE'::studies_type
+                   ELSE '1DEGREE'::studies_type
+        END
+    `);
```

Comment:
I bet you need to throw this into a `defer` call, otherwise adonis will be adonis and will run this during dry runs
(i should make a CI workflow for ensuring that dry runs not modifying the db)